content/en/docs/features/signed-images.md

```diff
@@ -8,4 +8,166 @@ tags:
 - images
 ---
 
-TODO
+
+
+# Overview
+[Encrypted images](/docs/features/encrypted-images/) provide confidentiality,
+but they do not provide _authenticity_ or _integrity_. Image signatures provide
+this additional property, preventing certain types of image tampering,
+[for example](https://docs.sigstore.dev/about/overview/#why-cryptographic-signing).
+
+In this brief guide, we show two tools that can be used to sign container images:
+[cosign](https://github.com/sigstore/cosign) and
+[skopeo](https://github.com/containers/skopeo). The skopeo tool can be
+used to to create both cosign signatures or "simple signatures" (which leverage
+gpg keys). For our purposes, our skopeo examples will use the simple signing
+approach. In any case, the general approach is to (1) create keys for signing,
+(2) sign a newly tagged image, and (3) update the
+[KBS](/docs/attestation/key-broker-service) with the public signature key and a
+security policy.
+
+# Creating an Image
+
+## Creating Keys
+
+### cosign
+To generate a public-private keypair with cosign, provide your
+`COSIGN_PASSWORD` and use the `generate-key-pair` action:
+```shell
+$ COSIGN_PASSWORD=just1testing2password3 cosign generate-key-pair
+```
+This will create the private and public keys: cosign.key and cosign.pub.
+
+### Simple Signing - gpg
+skopeo depends on gpg for a keypair.
+To generate a keypair with gpg using default options, use `--full-generate-key`:
+```shell
+$ gpg --full-generate-key
+```
+
+There are several prompts. A user for test purposes could be:
+```
+Github Runner
+git@runner.com
+just1testing2password3
+```
+
+Then export it. The `--export-secret-key` option is sufficient for exporting
+both the secret and public key. Example command:
+```shell
+$ gpg --export-secret-key F63DB2A1AB7C7F195F698C9ED9582CADF7FBCC5D > github-runner.keys
+```
+
+The keys can later be imported by gpg in a CI system using `--batch` to avoid
+typing the password:
+```shell
+$ gpg --batch --import ./github-runner.keys
+```
+
+When automating CI or test worfklows, you can place the password for the key in
+a plaintext file (when it is safe to do so):
+```shell
+echo just1testing2password3 > git-runner-password.txt
+```
+
+
+## Signing the Image
+### cosign
+
+Use the private key to sign an image.
+In this example, we assume that there is a Dockerfile (`your_dockerfile` below)
+for creating an image that you want signed. The workflow is to build the image,
+push it to ghcr (which requires `docker login`), and sign it.
+```shell
+COCO_PKG=confidential-containers/test-container
+docker build \
+  -t ghcr.io/$(COCO_PKG):cosign-sig \
+  -f <your_dockerfile> \
+  .
+docker push ghcr.io/$(COCO_PKG):cosign-sig
+cosign sign --key ./cosign.key ghcr.io/${COCO_PKG}:cosign-signed
```

Comment:
I think the tag `cosign-signed` should be `cosign-sig` .

Reply:
nice catch, changing